### DISCUSSION TITLE
Requiring signed commits

### DISCUSSION THREAD

**Discussion by MrBrain295** [mdn/mdn-community#133] [Content] [answered]
To be in accordance Yari I think the content repository should require signed commits and I wanted feedback. If this change is agreed upon I will be willing to make the changes in content.

**Comment by teoli2003** [comment]
To discuss this, we should have the following information:
- What would this bring to the content and to our users? 
- What problem would this solve?
- Will this prevent some people to contribute?

**Reply by MrBrain295** [reply]
This change would help prevent some malicious commits which would overall help the content repo. The problem this will help solve is the risk of malicious commits. This won’t prevent many people from contributing as most commits are made through GitHub.

**Comment by wbamberg** [comment]
I'm in favour of this. If I understand correctly, commits made through the GitHub UI are signed automatically? (I think you implied that in https://github.com/mdn/mdn-community/discussions/133#discussioncomment-2981200).

I think people using the CLI are likely to be set up for this already. At this point signing commits is such a common requirement that it's reasonable to expect it.

**Reply by MrBrain295** [reply]
The GitHub CLI does sign commits automatically ([see here](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification)).

**Reply by OnkarRuikar** [reply]
> The GitHub CLI does sign commits automatically

The [GitHub CLI](https://cli.github.com/manual/) has no command for commit. To auto sign commits while using [git](https://git-scm.com/), i.e. to avoid using `-S` flag on every commit, we need to explicitly configure it:
```
git config --global commit.gpgsign true
```

**Comment by teoli2003** [comment]
I am a bit afraid of the extra work that would come out of this.

There are three cases:
1. If a PR is created via the GitHub UI, no extra work.
2. If a PR is created using _git_ and commits in it are signed, no extra work.
3. If a PR is created using _git_ and one commit in it is not signed, a reviewer will need to explain to the author what to do it.

We have about 50 PRs a day and struggle to cope with them, so even only 3 can be a significant amount of additional work for reviewers.

**Reply by OnkarRuikar** [reply]
I agree. `mdn/content` has a lot of newbie contributors. And setting GPG for the *first* time can be a difficult task. And explaining the process will be even harder for the reviewers.

But it's a very basic [good to have requirement](https://security.stackexchange.com/a/120725). Everyone has to do the setup one day. Perhaps we can have a video on [Mozilla Youtube](https://www.youtube.com/user/Mozilla) channel about entire `mdn/content` dev environment setup. And provide the link/embed in  README.md and in the default text when we create a PR. @schalkneethling  wdyt?

**Reply by yin1999** [reply]
GitHub is just [supported SSH commit verification](https://github.blog/changelog/2022-08-23-ssh-commit-verification-now-supported/). I think it's an easier way to sign commits.

Steps to setup: <https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification#ssh-commit-signature-verification>

**Comment by schalkneethling** [comment, marked as answer]
I agree that the only way we can enforce this is if we have a very clear guide of how to set this up. I would also not make this a requirement from the first PR as not everyone would necessarily know about this depending on how they found the project. Once they _do_ contribute, we can enforce signed commits but, as just mentioned, only if we have a clear and easy to follow guide.

This is part of the work I will be doing in Q3 so, thank you for starting this discussion.
